src/CodeGen_LLVM.cpp

```diff
@@ -1195,6 +1204,94 @@ void CodeGen_LLVM::visit(const Cast *op) {
         value = builder->CreateUIToFP(value, llvm_dst);
     } else {
         internal_assert(src.is_float() && dst.is_float());
+
+        if (src.bits == 16 && dst.bits > 16 &&
+            target_needs_software_cast_from_float16_to(dst)) {
+            // Use software implementation for converting half to higher precision
+            // float when operating on scalar types. If the target doesn't support
+            // this natively LLVM will emit a call to ``__gnu_h2f_ieee()`` which isn't
+            // available as we don't use compiler-rt so we use our own
+            // implementation.
+            user_assert(src.width == 1 && src.width == dst.width) <<
+                "The target requires a software implementation of float16_t conversion"
+                " but vectorization has been requested which is not possible\n";
+            llvm::Function* convFunc = nullptr;
```

Comment:
Please use underscores instead of camelCase for local variables and functions throughout this PR.